src/praisonai/praisonai/auto.py

```
@@ -192,54 +165,45 @@ def _get_praisonai_tools():
-    if _openai_available is None:
-        try:
-            import openai  # noqa: F401
-            _openai_available = True
-        except ImportError:
-            _openai_available = False
-    return _openai_available
+    result = _load_optional("openai")
+    return result is not None
 
 
 def _get_openai_client(api_key: str = None, base_url: str = None):
```

Ruff (0.15.11)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)